### PR TITLE
fix migration controller error

### PR DESCRIPTION
**Pull request by ldpliu**

### CODE REVIEW

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go [hunk 1]
```diff
@@ -146,10 +146,11 @@ func (c *crdReconcile) updateStoredVersion(ctx context.Context) error {
 		// Check migration status before update CRD stored version
 		// If CRD's StorageVersionMigration is not found, it means that the previous or the current release CRD doesn't need migration, and can contiue to update the CRD's stored version.
 		// If CRD's StorageVersionMigration is found and the status is success, it means that the current CRs were migrated successfully, and can contiue to update the CRD's stored version.
-		// Other cases, for example, the migration failed, we should not contiue to update the stored version, that will caused the stored old version CRs inconsistent with latest CRD.
+		// Other cases, for example, the migration failed or not aplied, we should not contiue to update the stored version, that will caused the stored old version CRs inconsistent with latest CRD.
 		svmStatus, err := migrationcontroller.IsStorageVersionMigrationSucceeded(c.hubMigrationClient, name)
-		if svmStatus == false && !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to updateStoredVersion as StorageVersionMigrations %v: %v", name, err)
+		if !(svmStatus == true || errors.IsNotFound(err)) {
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go]: I am confused about this `if`. If is true when `svmStatus == false && !errors.IsNotFound(err)`, why it is different from before?

File: pkg/operators/clustermanager/controllers/migrationcontroller/migration_controller.go [hunk 2]
```diff
@@ -48,6 +48,10 @@ var (
 		"cluster-manager/cluster-manager-managedclustersets-migration.yaml",
 		"cluster-manager/cluster-manager-managedclustersetbindings-migration.yaml",
 	}
+	requiredMigrationResources = []string{
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/migrationcontroller/migration_controller.go:51]: what is the different between this and migrationRequestFiles?

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go [hunk 3]
```diff
@@ -146,10 +146,11 @@ func (c *crdReconcile) updateStoredVersion(ctx context.Context) error {
 		// Check migration status before update CRD stored version
 		// If CRD's StorageVersionMigration is not found, it means that the previous or the current release CRD doesn't need migration, and can contiue to update the CRD's stored version.
 		// If CRD's StorageVersionMigration is found and the status is success, it means that the current CRs were migrated successfully, and can contiue to update the CRD's stored version.
-		// Other cases, for example, the migration failed, we should not contiue to update the stored version, that will caused the stored old version CRs inconsistent with latest CRD.
+		// Other cases, for example, the migration failed or not aplied, we should not contiue to update the stored version, that will caused the stored old version CRs inconsistent with latest CRD.
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go]: typo: applied, cause

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go [hunk 4]
```diff
@@ -181,3 +196,34 @@ func (c *crdReconcile) updateStoredVersion(ctx context.Context) error {
...
 }
+
+// desiredStoredVersions caculate the desired stored versions
+// If the "requiredStoredVersion" do not exist, do not update the current stored versions
+// If the "requiredStoredVersion" exist, delete the "removedCRDStoredVersion"
+func desiredStoredVersions(curStoredVersion []string, requiredStoredVersion, removedCRDStoredVersion string) []string {
+	// remove old versions from its status
+	newStoredVersions := make([]string, 0, len(curStoredVersion))
+	if len(removedCRDStoredVersion) == 0 {
+		return curStoredVersion
```
Comment (haoqing0110) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go:207]: Should this function also return curStoredVersion here when len(requiredStoredVersion) == 0?

**Pull request closed without merging**